reframe/core/systems.py

```diff
@@ -13,6 +13,169 @@
 from reframe.core.environments import (Environment, ProgEnvironment)
 
 
+class Processor(jsonext.JSONSerializable):
+    '''A representation of a processor inside ReFrame.
+
+    .. warning::
+       Users may not create :class:`Processor` objects directly.
+    '''
+
+    def __init__(self, processor_info):
+        self._arch = None
+        self._num_cpus = None
```

Comment:
- `num_cpus_per_socket`: lscpu, papi_avail and likwid-topology are using `Cores per socket` -> can we use num_cores_per_socket ?

Reply:
The `num_cores_per_socket` value is already available as a derived property `num_cores // num_sockets`.

Reply:
@jgphpc The internal values of this will be the minimum required, from which we can derive other properties. I am strongly in favor of using `cpu` as the the name of any logical processing unit. This is consistent with Linux and I find it more precise than `thread`.